### PR TITLE
Rework completions section in docs

### PR DESCRIPTION
**Pull request by Secrus**
# Pull Request Check List

Resolves: #2295

- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: docs/_index.md [hunk 1]
```diff
@@ -282,42 +282,68 @@ To uninstall Poetry, simply delete the entire `$VENV_PATH` directory.
...
 
 ```bash
-# Bash
-poetry completions bash > /etc/bash_completion.d/poetry
+poetry completions bash >> ~/.bash_completion
+```
 
-# Fish
-poetry completions fish > ~/.config/fish/completions/poetry.fish
+#### Local, based on XDG directory spec
```
Comment (neersighted) [inline docs/_index.md:295]:
```suggestion
#### Lazy-loaded
```

File: docs/_index.md [hunk 2]
```diff
@@ -282,42 +282,68 @@ To uninstall Poetry, simply delete the entire `$VENV_PATH` directory.
 `poetry` supports generating completion scripts for Bash, Fish, and Zsh.
 See `poetry help completions` for full details, but the gist is as simple as using one of the following:
 
+### Bash
+
+There are few options in `bash`, choose the one that suits your workflow.
+
+#### Local, recommended
```
Comment (neersighted) [inline docs/_index.md:289]:
```suggestion
#### Auto-loaded (recommended)
```

File: docs/_index.md [hunk 3]
```diff
@@ -282,42 +282,68 @@ To uninstall Poetry, simply delete the entire `$VENV_PATH` directory.
...
-For `zsh`, you must then add the following lines in your `~/.zshrc`
+You must then add the following lines in your `~/.zshrc`
 
 ```bash
 fpath+=~/.zfunc
 autoload -Uz compinit && compinit
 ```
 
-For `oh-my-zsh`, you must then enable poetry in your `~/.zshrc` plugins
+#### Oh-My-Zsh
```
Comment (neersighted) [inline docs/_index.md:326]:
```suggestion
#### Oh My Zsh
```

File: docs/_index.md [hunk 4]
```diff
@@ -282,42 +282,68 @@ To uninstall Poetry, simply delete the entire `$VENV_PATH` directory.
...
-You may need to restart your shell in order for the changes to take effect.
-{{% /note %}}
+### Zsh
+
+```zsh
+poetry completions zsh > ~/.zfunc/_poetry
+```
 
-For `zsh`, you must then add the following lines in your `~/.zshrc`
+You must then add the following lines in your `~/.zshrc`
```
Comment (neersighted) [inline docs/_index.md:319]:
```suggestion
You must then add the following lines in your `~/.zshrc`, if they do not already exist:
```

File: docs/_index.md [hunk 5]
```diff
@@ -282,42 +282,68 @@ To uninstall Poetry, simply delete the entire `$VENV_PATH` directory.
...
 ```
+
+#### prezto
+
+```zsh
+poetry completions zsh > ~/.zprezto/modules/completion/external/src/_poetry
+```
+
+{{% note %}}
+You may need to restart your shell in order for the changes to take effect.
```
Comment (neersighted) [inline docs/_index.md:348]:
```suggestion
You may need to restart your shell in order for these changes to take effect.
```

File: docs/_index.md [hunk 6]
```diff
@@ -282,42 +282,68 @@ To uninstall Poetry, simply delete the entire `$VENV_PATH` directory.
...
-# Zsh
-poetry completions zsh > ~/.zfunc/_poetry
+```bash
+poetry completions bash > ${XDG_DATA_HOME:~/.local/share}/bash_completion/completions/poetry
+```
 
-# Oh-My-Zsh
-mkdir $ZSH_CUSTOM/plugins/poetry
-poetry completions zsh > $ZSH_CUSTOM/plugins/poetry/_poetry
+#### Global, might require root priviledges
```
Comment (neersighted) [inline docs/_index.md:301]: I would drop this entire section -- no point installing completions locally for a user-installed Poetry, and if Poetry is installed globally it's likely through distro packaging.

File: docs/_index.md [hunk 7]
```diff
@@ -282,42 +282,68 @@ To uninstall Poetry, simply delete the entire `$VENV_PATH` directory.
 `poetry` supports generating completion scripts for Bash, Fish, and Zsh.
 See `poetry help completions` for full details, but the gist is as simple as using one of the following:
 
+### Bash
+
+There are few options in `bash`, choose the one that suits your workflow.
```
Comment (neersighted) [inline docs/_index.md:287]: I think we can drop this line as well if we go with my simplification.

**Review by neersighted — CHANGES_REQUESTED** [review]
I'd like to see partial authorship attribution for the author of #4002 if that's appropriate (haven't compared in detail yet).

File: docs/_index.md (hunk 6 above)
Comment (Secrus) [reply docs/_index.md]: removed

File: docs/_index.md (hunk 7 above)
Comment (Secrus) [reply docs/_index.md]: sure, removed it.

**Comment by Secrus** [comment]
> I'd like to see partial authorship attribution for the author of #4002 if that's appropriate (haven't compared in detail yet).

@neersighted in this case, there is no code that was taken from that PR, nor was I inspired by it. I can add attribution, but I don't think it's appropriate in this case.

**Review by neersighted — APPROVED** [review]

**Pull request merged**